src/ondemand.coffee

```diff
@@ -231,119 +86,185 @@ module.exports = (robot) ->
+              else "#{md.codeMulti}Operator, there is no daily deal at the moment#{md.blockEnd}"
+
+            res.send message
+
   robot.respond /enemies/, (res) ->
     robot.logger.debug 'Entered persistent enemies command'
-    noEnemiesString = util.format '%sOperator, there is no sign of enemies, stay alert.%s', 
-                        dsUtil.codeMulti, dsUtil.blockEnd
     userDB.getPlatform res.message.room, (err, platform) ->
       if err
-          return robot.logger.error err
-        ws.getPersistentEnemies platform, (err, enemies) ->
-          if err 
-            res.send noEnemiesString
-            return robot.logger.error err
-          if enemies != ''
-            res.send util.format '%s%s%s', dsUtil.codeMulti, enemies, dsUtil.blockEnd
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getAllPersistentEnemiesString()), 300
+            
+  
+  robot.respond /event/, (res) ->
+    robot.logger.debug 'Entered events command'
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getEventsString()), 300
+  
+  robot.respond /invasions/, (res) ->
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        robot.logger.error err
+      else
+        ds.getInvasions platform, (err, data) ->
+          if err
+            robot.logger.error err
           else
-            res.send noEnemiesString
+            message =
+              if data.length then (invasion.toString() for invasion in data).join('\n\n')
+              else "#{md.codeMulti}Operator, there are no invasions at the moment#{md.blockEnd}"
+            res.send message  
+  robot.respond /news/, (res) ->
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getNewsString()), 300
+      
+  robot.respond /primeaccess/, (res) ->
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getPrimeAccessString()), 300
+  
+  robot.respond /update/, (res) ->
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getUpdatesString()), 300
+
   robot.respond /rewards/, (res) ->
-    res.send util.format('%sMission rewards%shttp://rewards.morningstar-wf.com%s',
-                           dsUtil.linkBegin, dsUtil.linkMid, dsUtil.linkEnd)
+    res.send "#{md.codeMulti}#{md.linkBegin}Mission rewards #{md.linkMid} http://rewards.morningstar-wf.com#{md.linkEnd}#{md.blockEnd}"
```

Comment:
Additional spaces break markdown formatting, should be `...rewards#{md.linkMid}http://...`